### PR TITLE
Config File Support

### PR DESCRIPTION
**Pull request by tgetgood**
Specified a config file format, example is `config.json.example` which is copied into the readme. Executing `name-your-contributors --config config.json` spits out a copy of the config file with an extra `contributions` key in each object which is what nyc would have spit out for that specific query.

Note, this is branched off of the cli-updates branch. I can rebase it after that gets merged if there are any conflicts. 

Addresses #29 

I can't attach json files, so here are the examples:

config format:

```json
{
  "token": "123435abcdf",
  "repos": [{
    "login": "mntnr",
    "repo": "name-your-contributors",
    "before": "2017-11-30",
    "after": "2017-06-01"
  }, {
    "login": "mntnr",
    "repo": "whodidwhat"
  }],
  "orgs": [{
    "login": "adventure-js",
    "after": "2017-07-01"
  }]
}
```

Return format:

```json
{
  "repos": [
    {
      "login": "mntnr",
      "repo": "name-your-contributors",
      "before": "2017-11-30",
      "after": "2017-06-01",
      "contributions": {
        "commitAuthors": [
          {
            "login": "RichardLitt",
            "name": "Richard Littauer",
            "url": "https://github.com/RichardLitt",
            "count": 42
          },
          {
            "login": "tgetgood",
            "name": "Thomas Getgood",
            "url": "https://github.com/tgetgood",
            "count": 100
          },
          {
            "login": "jozefizso",
            "name": "Jozef Izso",
            "url": "https://github.com/jozefizso",
            "count": 2
          }
        ],
        "commitCommentators": [],
        "prCreators": [
          {
            "login": "jozefizso",
            "name": "Jozef Izso",
            "url": "https://github.com/jozefizso",
            "count": 1
          },
          {
            "login": "greenkeeper",
            "count": 2
          },
          {
            "login": "RichardLitt",
            "name": "Richard Littauer",
            "url": "https://github.com/RichardLitt",
            "count": 2
          },
          {
            "login": "tgetgood",
            "name": "Thomas Getgood",
            "url": "https://github.com/tgetgood",
            "count": 7
          }
        ],
        "prCommentators": [
          {
            "login": "RichardLitt",
            "name": "Richard Littauer",
            "url": "https://github.com/RichardLitt",
            "count": 26
          },
          {
            "login": "tgetgood",
            "name": "Thomas Getgood",
            "url": "https://github.com/tgetgood",
            "count": 16
          }
        ],
        "issueCreators": [
          {
            "login": "RichardLitt",
            "name": "Richard Littauer",
            "url": "https://github.com/RichardLitt",
            "count": 14
          },
          {
            "login": "tgetgood",
            "name": "Thomas Getgood",
            "url": "https://github.com/tgetgood",
            "count": 5
          }
        ],
        "issueCommentators": [
          {
            "login": "RichardLitt",
            "name": "Richard Littauer",
            "url": "https://github.com/RichardLitt",
            "count": 21
          },
          {
            "login": "tgetgood",
            "name": "Thomas Getgood",
            "url": "https://github.com/tgetgood",
            "count": 15
          },
          {
            "login": "jywarren",
            "name": "Jeffrey Warren",
            "url": "https://github.com/jywarren",
            "count": 1
          },
          {
            "login": "diasdavid",
            "name": "David Dias",
            "url": "https://github.com/diasdavid",
            "count": 2
          }
        ],
        "reactors": [
          {
            "login": "RichardLitt",
            "name": "Richard Littauer",
            "url": "https://github.com/RichardLitt",
            "count": 8
          },
          {
            "login": "patcon",
            "name": "Patrick Connolly",
            "url": "https://github.com/patcon",
            "count": 1
          }
        ],
        "reviewers": [
          {
            "login": "RichardLitt",
            "name": "Richard Littauer",
            "url": "https://github.com/RichardLitt",
            "count": 19
          },
          {
            "login": "tgetgood",
            "name": "Thomas Getgood",
            "url": "https://github.com/tgetgood",
            "count": 7
          }
        ]
      }
    },
    {
      "login": "mntnr",
      "repo": "whodidwhat",
      "contributions": {
        "commitAuthors": [
          {
            "login": "RichardLitt",
            "name": "Richard Littauer",
            "url": "https://github.com/RichardLitt",
            "count": 3
          },
          {
            "login": "tgetgood",
            "name": "Thomas Getgood",
            "url": "https://github.com/tgetgood",
            "count": 10
          }
        ],
        "commitCommentators": [],
        "prCreators": [],
        "prCommentators": [],
        "issueCreators": [
          {
            "login": "tgetgood",
            "name": "Thomas Getgood",
            "url": "https://github.com/tgetgood",
            "count": 3
          },
          {
            "login": "RichardLitt",
            "name": "Richard Littauer",
            "url": "https://github.com/RichardLitt",
            "count": 3
          }
        ],
        "issueCommentators": [
          {
            "login": "RichardLitt",
            "name": "Richard Littauer",
            "url": "https://github.com/RichardLitt",
            "count": 7
          },
          {
            "login": "tgetgood",
            "name": "Thomas Getgood",
            "url": "https://github.com/tgetgood",
            "count": 6
          }
        ],
        "reactors": [
          {
            "login": "RichardLitt",
            "name": "Richard Littauer",
            "url": "https://github.com/RichardLitt",
            "count": 1
          }
        ],
        "reviewers": []
      }
    }
  ],
  "orgs": [
    {
      "login": "adventure-js",
      "after": "2017-07-01",
      "contributions": {
        "commitAuthors": [],
        "commitCommentators": [],
        "prCreators": [],
        "prCommentators": [],
        "issueCreators": [
          {
            "login": "RichardLitt",
            "name": "Richard Littauer",
            "url": "https://github.com/RichardLitt",
            "count": 2
          }
        ],
        "issueCommentators": [
          {
            "login": "noffle",
            "name": "Stephen Whitmore",
            "url": "https://github.com/noffle",
            "count": 2
          },
          {
            "login": "RichardLitt",
            "name": "Richard Littauer",
            "url": "https://github.com/RichardLitt",
            "count": 8
          },
          {
            "login": "reustle",
            "name": "Shane Reustle",
            "url": "https://github.com/reustle",
            "count": 1
          },
          {
            "login": "despairblue",
            "name": "Danny Martini",
            "url": "https://github.com/despairblue",
            "count": 2
          },
          {
            "login": "whyrusleeping",
            "name": "Whyrusleeping",
            "url": "https://github.com/whyrusleeping",
            "count": 2
          },
          {
            "login": "MylesBorins",
            "name": "Myles Borins",
            "url": "https://github.com/MylesBorins",
            "count": 1
          },
          {
            "login": "diffcunha",
            "name": "Diogo Cunha",
            "url": "https://github.com/diffcunha",
            "count": 1
          }
        ],
        "reactors": [
          {
            "login": "RichardLitt",
            "name": "Richard Littauer",
            "url": "https://github.com/RichardLitt",
            "count": 4
          },
          {
            "login": "despairblue",
            "name": "Danny Martini",
            "url": "https://github.com/despairblue",
            "count": 3
          },
          {
            "login": "yangwao",
            "name": "Matej Nemček",
            "url": "https://github.com/yangwao",
            "count": 2
          },
          {
            "login": "noffle",
            "name": "Stephen Whitmore",
            "url": "https://github.com/noffle",
            "count": 1
          },
          {
            "login": "carlesba",
            "name": "Carles Ballester",
            "url": "https://github.com/carlesba",
            "count": 1
          },
          {
            "login": "fbaiodias",
            "name": "Francisco Baio Dias",
            "url": "https://github.com/fbaiodias",
            "count": 1
          },
          {
            "login": "fforres",
            "name": "Felipe Torres (fforres)",
            "url": "https://github.com/fforres",
            "count": 1
          },
          {
            "login": "whyrusleeping",
            "name": "Whyrusleeping",
            "url": "https://github.com/whyrusleeping",
            "count": 1
          }
        ],
        "reviewers": []
      }
    }
  ]
}
```

### CODE REVIEW

**Comment by RichardLitt** [comment]
- Rebase.
- In the config, change `login` to `user`. I think that makes more sense.
- Add a config example somewhere - in the README should work.

Otherwise, looks great!

**Comment by tgetgood** [comment]
I put some config examples in the README. Which part isn't clear? I'll elaborate on it.

As for login vs user, I was going for uniformity between repos and orgs. At the API level, they're both logins, but they're not both users. 

We could do `user` for repos and `org` or `orgName` or just `name` for orgs. 

How about 

```json
{
  "token": "123435abcdf",
  "repos": [{
    "user": "mntnr",
    "repo": "name-your-contributors",
    "before": "2017-11-30",
    "after": "2017-06-01"
  }, {
    "user": "mntnr",
    "repo": "whodidwhat"
  }],
  "orgs": [{
    "name": "adventure-js",
    "after": "2017-07-01"
  }]
}
```

File: src/cli.js [hunk 1]
```diff
@@ -34,6 +16,8 @@ const cli = meow([`
     -r, --repo   - Repository to search
     -t, --token  - GitHub auth token to use
     -u, --user   - User to which repository belongs
+    --config     - Operate from config file. In this mode only token, verbose, and
```
Comment (RichardLitt) [inline src/cli.js]: move the -c alias to config, it is more likely to be used than --csv

**Comment by RichardLitt** [comment]
Ah, missed that, sweet. 

Hmm. If that is how GitHub's API does it, lets just stay with login. It works for both.

**Comment by tgetgood** [comment]
Sorry about the mangled rebase. Fixed now.

File: .travis.yml [hunk 2]
```diff
@@ -6,6 +6,7 @@ cache:
 notifications:
   email: false
 node_js:
+  - 'node'
```
Comment (RichardLitt) [inline .travis.yml]: Remove. Will break tests, unfortunately.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 Please contribute! Here are some things that would be great:
 
-- [Open an issue!](https://github.com/mntnr/name-your-contributors/issues/new)
+- [Open an issue!](https://github.com/RichardLitt/name-your-contributors/issues/new)
```
Comment (RichardLitt) [inline CONTRIBUTING.md]: Shouldn't be changed.

**Comment by RichardLitt** [comment]
Not sure it's fixed? There's some reverts which are odd.

**Comment by tgetgood** [comment]
Was not fixed. This looks better.

File: readme.md [hunk 4]
```diff
@@ -79,6 +79,57 @@ $ export GITHUB_TOKEN={your-token}
...
+```json
+{
+  "repos": [{
+	"login": "mntnr",
+	"repo": "name-your-contributors",
+	"before": "2017-11-30",
+	"after": "2017-06-01",
+	"contributions" : {
+	  "commitAuthors": [...],
+	  "commitCommentators: [...],
```
Comment (RichardLitt) [inline readme.md]: Missing closing "

Comment (RichardLitt) [reply readme.md]: I got it.

**Comment by RichardLitt** [comment]
Woop woop!

**Pull request merged**